src/components/molecules/NavBar/playa.scss

```diff
@@ -60,12 +59,13 @@ $sand: #937c63;
     }
 
     .navbar-links {
-      padding: 12px 20px;
+      padding: 11px 20px;
       display: flex;
       flex-wrap: wrap;
       align-items: center;
       align-content: space-between;
       justify-content: flex-end;
+      background-color: inherit;
```

Comment:
Why do we need to change this here? Presumably the `.navbar-links` are rendered within the `.navbar-container`, which explicitly sets a background color. The default is 'transparent', so that it will show the same background as the element behind it. Is that not sufficient here?

- https://developer.mozilla.org/en-US/docs/Web/CSS/background-color#formal_definition

Reply:
Let me explain the logic of the fix.

Here is the html structure.
```html
<div class="navbar">                        <--- main navbar container z-index: 30
  <div class="nav-container">           <--- background: black;
    <div class="navbar-links">
      <div class="NavSearchBar">
        <div class="NavSearchBar__nav-dropdown" />
        <div class="input-field" />
      </div>
    </div>
  </div>
</div>

...
<div class="schedule-dropdown-backdrop" />
```

The fix consists of 2 parts: 
1. `background-color: inherit`
initially divs have transparent color by default, that's why even being behind those divs, `NavSearchBar__nav-dropdown` was visible anyway. (see the screenshot of its appearing from the top)

![image](https://user-images.githubusercontent.com/63313732/116518545-605def00-a89e-11eb-9516-d11d3e59ef59.png)

specifying `inherit` background-color we tell `navbar-links` to take the background-color of its parent (e.g. black), so it's not transparent anymore and elements can be hidden behind that.

![image](https://user-images.githubusercontent.com/63313732/116519086-0d386c00-a89f-11eb-87e5-96ca5917f4bf.png)

2. `z-index: -1`
`navbar` creates a stacking context because it has `z-index`. 

* https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context
![image](https://user-images.githubusercontent.com/63313732/116520280-86848e80-a8a0-11eb-9d27-d092b66fe5ac.png)


it means that if we apply `z-index` to its children, the meaning of that "child" z-index will be within "parent" z-index only.
So, if we apply `z-index: 20` to `NavSearchBar__nav-dropdown` which is lower value than `navbar` has (30)
it won't hide `NavSearchBar__nav-dropdown` behind `navbar` because its z-index works within parent stacking context.

That's why I use `z-index: -1` to have it behind all the elements within `navbar`.

That also explains why we can't use same z-index as `schedule-dropdown-backdrop` has.
You can notice that `navbar` and `schedule-dropdown-backdrop` don't have child-parent relationships. That's why their z-indexes work as expected.

Reply:
> that's why even being behind those divs, NavSearchBar__nav-dropdown was visible anyway

@sunny-viktoryia In that screenshot it doesn't look like it's behind the NavBar (eg. `z-index`) though?

![image](https://user-images.githubusercontent.com/753891/116527153-b8880600-a91d-11eb-831b-5e2be06de96b.png)

---

> So, if we apply z-index: 20 to NavSearchBar__nav-dropdown which is lower value than navbar has (30) it won't hide NavSearchBar__nav-dropdown behind navbar because its z-index works within parent stacking context.

@sunny-viktoryia Is the 'drawer' not positioned `absolute`/similar that removes it from the parent stacking context?

---

> That also explains why we can't use same z-index as schedule-dropdown-backdrop has.
> You can notice that navbar and schedule-dropdown-backdrop don't have child-parent relationships. > That's why their z-indexes work as expected.

@sunny-viktoryia Makes sense. What would it take to remove the child-parent relationship here so that it can work in the same way? As that seems like it would likely be the better overall fix here?

---

@mike-lvov Thoughts?

Reply:
> In that screenshot it doesn't look like it's behind the NavBar (eg. z-index) though?

you're right. It's not behind NavBar (as it's in its stacking context), but it's behind transparent `.navbar-links`. And the 2nd 
screenshot was made when background-color of `.navbar-links` was set to `inherit` (not transparent, but black) - that was the main idea of the fix.

![](https://user-images.githubusercontent.com/63313732/116519086-0d386c00-a89f-11eb-87e5-96ca5917f4bf.png)

> What would it take to remove the child-parent relationship here so that it can work in the same way? As that seems like it would likely be the better overall fix here?

that what I tried to clarify in the #design slack channel yesterday. 
As far as I see it, we'll have to change the structure of the navbar components. 
It seems that we'll have to leave input-field as the child of the NavBar component and move search-results dropdown out of it.

... looking into the code to understand how the schedule opens ... 

NavBar component (`src/components/molecules/NavBar/NavBar.tsx`) is pretty big, and it has not only `.navbar` but `SchedulePageModal` there as well. So we can move schedule-results there as a sibling of `.navbar` and `shedule`. and using callback from NavSearchBar (on search query change) we can pass props to 'schedule-results'.

(note) 1 possible thing we need to think about is about positioning. When input and search-results are at the same component it's not difficult to place them vertically one under another , but as there may (or not) be radio (and maybe other components, which I don't know about) the positioning can differ.

Reply:
@0xdevalias @sunny-viktoryia 
I think the perfect way to do it would be to get search options out of the navbar stacking context and show them behind the navbar itself, but since the fix works as expected & doesn't have any drawbacks(or I just can't think of any?), let's leave this fix as is. I don't mind having it as OOS for every navbar element.

Let's focus on delivering OHBM features

Reply:
@mike-lvov @sunny-viktoryia  I'm ok with that, but can we we please add `@debt` comments to the places where this 'hack fix' was added that call this out so that we have context for future refactors to fix it properly?

Reply:
added the comments: https://github.com/sparkletown/sparkle/pull/1350/commits/d2e5fe9ef30c6cf6333f10fe908a91edb376ffc5